hrjobcontract/js/src/job-contract/controllers/contract.js

```diff
@@ -69,25 +69,17 @@ define([
             };
 
             function updateContractList(newEndDate){
-                var isCurrentContract;
-
-                //Is contract end date changed
-                if ($scope.details.period_end_date ?
-                new Date($scope.details.period_end_date).getTime() !== new Date(newEndDate).getTime() :
-                !!$scope.details.period_end_date !== !!newEndDate) {
-
-                    isCurrentContract = !newEndDate || new Date(newEndDate).setHours(0,0,0,0) >= new Date().setHours(0,0,0,0);
-
-                    if (isCurrentContract != !!+$scope.$parent.contract.is_current) {
-                        if (isCurrentContract) {
-                            $scope.$parent.contract.is_current = '1';
-                            $scope.$parent.$parent.contractCurrent.push($scope.$parent.contract);
-                            $scope.$parent.$parent.contractPast.splice($scope.$parent.$parent.contractPast.indexOf($scope.$parent.contract),1);
-                        } else {
-                            $scope.$parent.contract.is_current = '0';
-                            $scope.$parent.$parent.contractPast.push($scope.$parent.contract);
-                            $scope.$parent.$parent.contractCurrent.splice($scope.$parent.$parent.contractCurrent.indexOf($scope.$parent.contract),1)
-                        }
+                var isCurrentContract = !newEndDate || new Date(newEndDate).setHours(0,0,0,0) >= new Date().setHours(0,0,0,0);
+
+                if (isCurrentContract != !!+$scope.$parent.contract.is_current) {
+                    if (isCurrentContract) {
+                        $scope.$parent.contract.is_current = '1';
+                        $scope.$parent.$parent.contractCurrent.push($scope.$parent.contract);
+                        $scope.$parent.$parent.contractPast.splice($scope.$parent.$parent.contractPast.indexOf($scope.$parent.contract),1);
+                    } else {
+                        $scope.$parent.contract.is_current = '0';
+                        $scope.$parent.$parent.contractPast.push($scope.$parent.contract);
```

Comment:
This `$scope.$parent{1,}` is just awful. I remember encountering some time ago and in the end found out that the `$scope` already inherits all the data of the parent scope without having to access the `$parent` object, so in theory `$scope.contractPast` should work just as fine.

Could you check if that's true? Can we make this code better? Surely there's must be a better approach rather than accessing the `$parent` object an arbitrary number of times